zio-http/src/main/scala/zio/http/ChannelEvent.scala

```diff
@@ -18,49 +18,28 @@ package zio.http
 
 /**
  * Immutable and type-safe representation of events that are triggered on a
- * netty channel. `A` represents the inbound message type and `B` represents the
- * outbound message type that's allowed on the channel.
+ * netty channel. `A` represents the message type.
  */
-final case class ChannelEvent[-A, +B](channel: Channel[A], event: ChannelEvent.Event[B]) { self =>
-  def contramap[A1](f: A1 => A): ChannelEvent[A1, B] = copy(channel = channel.contramap(f))
-  def map[B1](f: B => B1): ChannelEvent[A, B1]       = copy(event = event.map(f))
-}
-
-object ChannelEvent {
-  def channelRead[A, B](channel: Channel[A], msg: B): ChannelEvent[A, B] =
-    ChannelEvent(channel, ChannelRead(msg))
-
-  def channelRegistered[A](channel: Channel[A]): ChannelEvent[A, Nothing] =
-    ChannelEvent(channel, ChannelRegistered)
-
-  def channelUnregistered[A](channel: Channel[A]): ChannelEvent[A, Nothing] =
-    ChannelEvent(channel, ChannelUnregistered)
+sealed trait ChannelEvent[+A] { self =>
+  import ChannelEvent._
 
-  def exceptionCaught[A](channel: Channel[A], cause: Throwable): ChannelEvent[A, Nothing] =
-    ChannelEvent(channel, ExceptionCaught(cause))
-
-  def userEventTriggered[A](channel: Channel[A], evt: UserEvent): ChannelEvent[A, Nothing] =
-    ChannelEvent(channel, UserEventTriggered(evt))
-
-  /**
-   * Immutable and type-safe representation of events that are triggered on a
-   * channel.
-   */
-  sealed trait Event[+A] { self =>
-    def map[B](f: A => B): Event[B] = self match {
-      case ChannelRead(msg)              => ChannelRead(f(msg))
-      case ChannelRegistered             => ChannelRegistered
-      case ChannelUnregistered           => ChannelUnregistered
+  def map[B](f: A => B): ChannelEvent[B] =
```

Comment:
In the `match` statement below there is only `Read` shows a different different from identity. Maybe it would be to make the default implementation as identity function and override it for `Read` only?

Or make implementation like
```
    def map[B](f: A => B): Event[B] = self match {
      case ChannelRead(msg)              => ChannelRead(f(msg))
      case other                                     => other
    }
```
?

Or I am completely wrong and Scala compiler optimises this statement and it won't check all 5 conditions in the worst case?